File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> mp;
        for (int i = 0; i < nums.size(); i++)
        {
            int diff = target - nums[i];
            if (mp.count(diff)) return {mp[diff]-1, i};
            mp[nums[i]] = i+1;
        }

        return {-1, -1};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.go
func twoSum(numbers []int, target int) []int {
	l, r := 0, len(numbers)-1
	for ; l < r; {
		tot := numbers[l] + numbers[r]
		if tot == target{
			return []int{l+1, r+1}
		}
		if tot < target{
			l++
		}else{
			r--
		}
	}
	return []int{-1, -1}
}
